application/views.py

```diff
@@ -1,37 +1,43 @@
 from django.http import HttpResponseRedirect
 from django.shortcuts import render
-from .forms import HighSchoolApplicationForm
-from .models import HighSchoolApplication
 from django.utils import timezone
 from django.urls import reverse
 
+from .forms import HighSchoolApplicationForm
+from .models import HighSchoolApplication
+from high_school.models import Program
+
 
 def new_application(request, user_type):
-    if request.method == "POST":
-        # TODO user_id will be replaced by sessions
-        user_id = 1
-        form = HighSchoolApplicationForm(request.POST)
-        if form.is_valid():
-            f = form.save(commit=False)
-            f.user_id = user_id
-            f.application_number = (
-                str(user_id) + str(f.school.school_name) + str(f.program)
-            )
-            f.submitted_date = timezone.now()
-            try:
-                if "Submit" in request.POST.get("submit"):
+    try:
+        if request.method == "POST":
+            # TODO user_id will be replaced by sessions
+            user_id = 1
+            form = HighSchoolApplicationForm(request.POST)
+            if form.is_valid():
+                f = form.save(commit=False)
+                f.user_id = user_id
+                f.application_number = generate_application_number(
+                    user_id, f.school.dbn, f.program.pk
+                )
+                if HighSchoolApplication.objects.filter(
+                    application_number=f.application_number
+                ):
+                    raise ValueError("Duplicate school and program selected")
```

Comment:
Would you be able to elaborate a bit more on how are we handling this case in the UI.

Reply:
I have not yet added an Add button. If an application has already been created and lets say you are creating a new application with same school n prog. It will give the error like this.

Reply:
But will we ever be creating more than 1 applications? So, what I wanted to convey earlier was, say for some weird reason this does happen...we should be handling this in the UI so that the user is properly notified.

Reply:
Of course, the current flow of saving into draft and submitting it individually is still there. Once i add the Add button i will remove the drafting. Adding multiple fields for school n program in the same form is very tricky, so i have currently the kept the form in the same way which was before.

Reply:
Okay, I think that answers my query.